tests/framework/project/test_pipeline_registry.py

```diff
@@ -25,7 +25,15 @@ def register_pipelines():
     sys.path.pop(0)
 
 
-def test_pipelines_without_configure_project_is_empty():
+def test_pipelines_without_configure_project_is_empty(
+    mock_package_name_with_pipelines_file,  # pylint: disable=unused-argument
+):
+    # Reimport `pipelines` from `kedro.framework.project` to ensure that
+    # it was not set by a pior call to the `configure_project` function.
+    del sys.modules["kedro.framework.project"]
```

Comment:
I remember spending a LONG time pinning down a very mysterious bug with tests a while ago which resulted in this bit of code:
https://github.com/kedro-org/kedro/blob/59064403e2590c992dceb51942f7dd9f6c97898a/tests/framework/cli/conftest.py#L146-L154

I think you're doing the same thing here and on L37 of `test_pipeline_discovery`. Do you feel ok with this or do we need some sort of `autouse` fixture throughout the whole test suite to make sure this doesn't trip us up anywhere else in future?

Reply:
I don't think I'm familiar enough with the broader set of issues related to this to feel confident about adding it without more thinking. I'm not sure if this could happen, but I'm also worried in case our tests pass because we unload everything but some actual scenarios that shouldn't pass work (not sure this would actually be possible).

Should we perhaps create a new issue to look into this, or alternatively try to solve it when it happens a third time?

Reply:
I think creating a new issue to look at this separately would be the best idea.

Reply:
#1787 created